cpp/dolfinx/fem/interpolate.h

```diff
@@ -1164,25 +1104,122 @@ create_nonmatching_meshes_interpolation_data(const mesh::Mesh<T>& mesh0,
       mesh0.geometry(), element0, mesh1, cells, padding);
 }
 
+/// @brief  Interpolate a finite element function one one grid to a function
+/// defined on another (non-matching) grid.
+/// @tparam T The Function scalar type
+/// @tparam U The Mesh geometry scalar type
+/// @param u The function to interpolate into
+/// @param v The function to interpolate from
+/// @param padding Absolute padding of bounding boxes of all entities on the
+/// mesh of \p u
+/// @note For repeated interpolation, it is adviced to precompute the
+/// non-matching interpolation data with @ref
+/// create_nonmatching_meshes_interpolation_data
+template <dolfinx::scalar T, std::floating_point U>
+void interpolate_nonmatching_meshes(Function<T, U>& u, const Function<T, U>& v,
+                                    T padding)
+{
+  assert(u.function_space());
+  const auto& mesh0 = u.function_space()->mesh();
+  int tdim = mesh0.topology()->dim();
+  auto cell_map = mesh0.topology()->index_map(tdim);
+  assert(cell_map);
+  std::int32_t num_cells = cell_map->size_local() + cell_map->num_ghosts();
+  std::vector<std::int32_t> cells(num_cells, 0);
+  std::iota(cells.begin(), cells.end(), 0);
+  const auto nmm_interpolation_data
+      = create_nonmatching_meshes_interpolation_data(
+          u->function_space()->mesh(), u->function_space()->element(),
+          v->function_space()->mesh(), cells, padding);
+  interpolate_nonmatching_meshes(u, v, cells, nmm_interpolation_data);
+}
+
+/// @brief  Interpolate a finite element function one one grid to a function
+/// defined on another (non-matching) grid.
+/// @tparam T The Function scalar type
+/// @tparam U The Mesh geometry scalar type
+/// @param u The function to interpolate into
+/// @param v The function to interpolate from
+/// @param cells THe cells in the mesh associated with @p u that will be
+/// interpolated into
+/// @param nmm_interpolation_data Data required for associating the
+/// interpolation points of @p u with cells in @p v
+template <dolfinx::scalar T, std::floating_point U>
+void interpolate_nonmatching_meshes(
+    Function<T, U>& u, const Function<T, U>& v,
+    std::span<const std::int32_t> cells,
+    const std::tuple<std::span<const std::int32_t>,
```

Comment:
This data structure should really be made a `struct` - would be much clearer and could be documented properly.